client/src/components/dashboard/DashboardEditor.jsx

```diff
@@ -0,0 +1,262 @@
+import React, { Component, PropTypes } from 'react';
+import ReactGridLayout from 'react-grid-layout';
+import DashboardVisualisationList from './DashboardVisualisationList';
+import DashboardCanvasItem from './DashboardCanvasItem';
+
+require('../../styles/DashboardEditor.scss');
+require('../../../node_modules/react-grid-layout/css/styles.css');
+require('../../../node_modules/react-resizable/css/styles.css');
+
+const getArrayFromObject = (object) => {
+  const arr = [];
+
+  Object.keys(object).forEach((key) => {
+    const item = object[key];
+
+    arr.push(item);
+  });
+
+  return arr;
+};
+
+const getNewEntityId = (entities, itemType) => {
+  const entityArray = getArrayFromObject(entities);
+  let highestIdInt = 0;
+
+  entityArray.forEach(item => {
+    if (item.type === itemType) {
+      const idInt = parseInt(item.id.substring(itemType.length + 1), 10);
+      if (idInt > highestIdInt) highestIdInt = idInt;
+    }
+  });
+
+  const newIdInt = highestIdInt + 1;
+
+  return `${itemType}-${newIdInt}`;
+};
+
+const getFirstBlankRowGroup = (layout, height) => {
+  /* Function to find the first collection of blank rows big enough for the
+  /* default height of the entity about to beinserted. */
+
+  /* If layout is empty, return the first row */
+  if (layout.length === 0) return 0;
+
+  const occupiedRows = {};
+  let lastRow = 0;
+
+  /* Build an object of all occupied rows, and record the last currently
+  /* occupied row. */
+  layout.forEach(item => {
+    for (let row = item.y; row < (item.y + item.h); row++) {
+      occupiedRows[row] = true;
+      if (row > lastRow) lastRow = row;
+    }
+  });
+
+  /* Loop through every row from 0 to the last occupied. If we encounter a blank
+  /* row i, check the next sequential rows until we have enough blank rows to
+  /* fit our height. If we do, return row i. */
```

Comment:
To be honest this is just how Sublime Text 2 auto-wraps long comments - but this is probably configurable somewhere, I'll take a look